src/helpers/distanceMatrix.js

```diff
@@ -1,22 +1,40 @@
 const get = require('lodash/get');
 const DistanceMatrix = require('../models/DistanceMatrix');
 const maps = require('../models/Google/Maps');
+const { TRANSIT, WALKING, PUBLIC_TRANSPORT } = require('./constants');
 
 exports.getDistanceMatrices = async credentials =>
   DistanceMatrix.find({ company: get(credentials, 'company._id') }).lean();
 
+const isDistanceMatrixDefine = (res) => {
+  if (res.status !== 200 || !res.data.rows[0] || !res.data.rows[0].elements ||
+    !res.data.rows[0].elements[0].duration || !res.data.rows[0].elements[0].distance) {
+    return false;
+  }
+
+  return true;
+};
+
 exports.getOrCreateDistanceMatrix = async (params, companyId) => {
-  const distanceMatrix = await DistanceMatrix.findOne(params);
+  let res = null;
+  if (params.mode === TRANSIT || params.mode === PUBLIC_TRANSPORT) {
+    const queryWithTransit = { ...params, mode: TRANSIT, key: process.env.GOOGLE_CLOUD_PLATFORM_API_KEY };
+    const resWithTransit = await maps.getDistanceMatrix(queryWithTransit);
+    const queryWithWalking = { ...params, mode: WALKING, key: process.env.GOOGLE_CLOUD_PLATFORM_API_KEY };
+    const resWithWalking = await maps.getDistanceMatrix(queryWithWalking);
```

Comment:
on pourrait faire un promise.all non ? il peut y avoir des interferences ?

Reply:
comment fais-tu ton promise.all sachant qu'il faut assigner le retour à une constante ? je suis pas sûr de comprendre

Reply:
Je suis pas du tout calé en asynchrone mais d'apres 
https://stackoverflow.com/questions/59477505/set-value-of-global-variable-inside-async-function-or-inside-promise
tu peux t'en sortir
```

(async () => {
    var promiseTransit = new Promise((resolve, reject) =>  {
        const res = await maps.getDistanceMatrix(queryWithTransit)(query)
        if (res.isPositive) resolve(res.value);
        if (res.isNegative) reject(false)
    });
    let resWithTransit="";
    const thenedPromiseTransit = promiseTransit.then((value) {
     resWithTransit=value;
    });

// meme chose avec Walking

    await Promise.all([thenedPromiseTransit,  thenedPromiseWalking]);

Reply:
Je trouve la méthode d'Ulysse plus compréhensible, sachant qu'on y gagne que très peu de temps.
Tu voyais d'autres avantages ?

Reply:
non